lib/features/app_terms/domain/set_newest_accepted_app_terms_usecase.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:zachranobed/common/logger/zo_logger.dart';
+import 'package:zachranobed/features/app_terms/domain/update_current_user_data_use_case.dart';
+import 'package:zachranobed/features/appConfiguration/data/get_last_app_terms_version_use_case.dart';
+
+class SetNewestAcceptedAppTermsUseCase {
+  final UpdateCurrentUserDataUseCase _updateCurrentUserDataUseCase;
+  final GetLastAppTermsVersionUseCase _getLastAppTermsVersionUseCase;
+
+  SetNewestAcceptedAppTermsUseCase(this._updateCurrentUserDataUseCase, this._getLastAppTermsVersionUseCase);
+
+  Future<void> setNewestAcceptedAppTerms() async {
+    int? lastAppTermsVersion = await _getLastAppTermsVersionUseCase.getLastAppTerms();
+
+    if (lastAppTermsVersion != null) {
+      _updateCurrentUserDataUseCase.updateCurrentUserData({
+        "lastAcceptedAppTermsVersion": lastAppTermsVersion
+      });
```

Comment:
I would pass only an integer, as domain should know nothing how data will be send

Reply:
Makes sense, changed